File: plugins/OrchidPosts/src/main/java/com/eden/orchid/posts/menu/TagsMenuType.java
package com.eden.orchid.posts.menu;

import com.eden.common.util.EdenPair;
import com.eden.common.util.EdenUtils;
import com.eden.orchid.api.OrchidContext;
import com.eden.orchid.api.theme.menus.menuItem.OrchidMenuItem;
import com.eden.orchid.api.theme.menus.menuItem.OrchidMenuItemFactory;
import com.eden.orchid.api.theme.pages.OrchidPage;
import com.eden.orchid.posts.PostsModel;
import com.eden.orchid.posts.pages.PostPage;
import com.eden.orchid.posts.pages.PostTagArchivePage;
import org.json.JSONObject;

import javax.inject.Inject;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class TagsMenuType implements OrchidMenuItemFactory {

    private OrchidContext context;
    private PostsModel postsModel;

    @Inject
    public TagsMenuType(OrchidContext context, PostsModel postsModel) {
        this.context = context;
        this.postsModel = postsModel;
    }

    @Override
    public List<OrchidMenuItem> getMenuItems(JSONObject menuItemJson) {
        List<OrchidMenuItem> items = new ArrayList<>();

        if(!EdenUtils.isEmpty(postsModel.getTagNames())) {
            List<OrchidPage> pages = new ArrayList<>();
            for(Map.Entry<String, EdenPair<List<PostPage>, List<PostTagArchivePage>>> tag : postsModel.getTags().entrySet()) {
                pages.add(tag.getValue().second.get(0));
            }
            items.add(new OrchidMenuItem(context, "Tags", pages));
        }

        return items;
    }
}

File: plugins/OrchidPosts/src/main/kotlin/com/eden/orchid/posts/menu/TagsMenuType.kt
package com.eden.orchid.posts.menu

import com.eden.common.util.EdenUtils
import com.eden.orchid.api.OrchidContext
import com.eden.orchid.api.theme.menus.menuItem.OrchidMenuItem
import com.eden.orchid.api.theme.menus.menuItem.OrchidMenuItemImpl
import com.eden.orchid.api.theme.pages.OrchidPage
import com.eden.orchid.posts.model.PostsModel
import java.util.*
import javax.inject.Inject

class TagsMenuType @Inject
constructor(context: OrchidContext, private val postsModel: PostsModel) : OrchidMenuItem(context, "postTags", 100) {

    override fun getMenuItems(): List<OrchidMenuItemImpl> {
        val items = ArrayList<OrchidMenuItemImpl>()

        if (!EdenUtils.isEmpty(postsModel.tagNames)) {
            val pages = ArrayList<OrchidPage>()
            for ((_, value) in postsModel.tags) {
                pages.add(value.second[0])
            }
            items.add(OrchidMenuItemImpl(context, "Tags", pages))
        }

        return items
    }
}

